src/app/rundown-view/components/rundown-header-context-menu/rundown-header-context-menu.component.ts

```diff
@@ -2,54 +2,82 @@ import { ChangeDetectionStrategy, Component, Input } from '@angular/core'
 import { RundownService } from '../../../core/abstractions/rundown.service'
 import { Rundown } from '../../../core/models/rundown'
 import { DialogService } from '../../../shared/services/dialog.service'
+import { ContextMenuOption } from '../../../shared/abstractions/context-menu-option'
 
 @Component({
   selector: 'sofie-rundown-header-context-menu',
   templateUrl: './rundown-header-context-menu.component.html',
-  styleUrls: ['./rundown-header-context-menu.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class RundownHeaderContextMenuComponent {
   @Input()
-  public rundown?: Rundown
+  public rundown: Rundown
+
+  @Input()
+  public shortenedRundownName: string
+
+  private readonly contextMenuOptionsForInactiveRundown: ContextMenuOption[] = [
+    {
+      label: 'Activate (On Air)',
+      contextAction: (): void => this.openActivationDialog(),
+    },
+    {
+      label: 'Reset Rundown',
+      contextAction: (): void => this.resetRundown(),
+    },
+  ]
+
+  private readonly contextMenuOptionsForActiveRundown: ContextMenuOption[] = [
+    {
+      label: 'Deactivate',
+      contextAction: (): void => this.openDeactivationDialog(),
+    },
+    {
+      label: 'Take',
+      contextAction: (): void => this.take(),
+    },
+    {
+      label: 'Reset Rundown',
+      contextAction: (): void => this.resetRundown(),
+    },
+  ]
+
+  public get contextMenuOptions(): ContextMenuOption[] {
+    return this.rundown.isActive ? this.contextMenuOptionsForActiveRundown : this.contextMenuOptionsForInactiveRundown
+  }
 
   constructor(
     private readonly rundownService: RundownService,
     private readonly dialogService: DialogService
   ) {}
 
   public openActivationDialog(): void {
-    if (!this.rundown || this.rundown.isActive) {
+    if (this.rundown.isActive) {
```

Comment:
At what point is this check performed, while the rundown is active?

Reply:
Now that i think a bit about it, it might be needed, incase the Rundown is activated via another frontend.

Reply:
Another frontend should trigger update of the rundown entity used here, so this check should be unnecessary. I have removed the check and tested that it still works.

Reply:
Testing on develop, and found that the context menu updates if the rundown active state changes while it is open, thus the if check should not be needed, ifi have my logic right.